samples/net/zperf/src/zperf_tcp_receiver.c

```diff
@@ -140,21 +145,34 @@ static void zperf_tcp_rx_thread(const struct shell *shell, int port)
 			return;
 		}
 
-		ret = zperf_get_ipv4_addr(shell, MY_IP4ADDR,
-					  &in4_addr_my->sin_addr);
-		if (ret < 0) {
-			shell_fprintf(shell, SHELL_WARNING,
-				      "Unable to set IPv4\n");
-			return;
+		if (MY_STATIC_IP4ADDR) {
```

Comment:
As the CONFIG_NET_CONFIG_MY_IPV4_ADDR is a string then the value can be an empty string. In that case this check would succeed even if the user did not placed any IP address there.
Better to do the check like this
```
if (MY_STATIC_IP4ADDR && MY_STATIC_IP4ADDR[0]) {
...
}
```

Reply:
It seems to be allowed empty string. Because this string will be set to iface in zperf_get_ipv4_addr.
This check could be done by net_if_addr_init.  Empty string is a type of wrong string. Do you agree?

Reply:
Well, in a way empty string is the same as not setting the IP address at all. So in that case you could select the default interface and IP address.
BTW, if you know the target IP address, you could use the `net_if_ipv4_select_src_addr()` and `net_if_ipv6_select_src_addr()` functions to select the proper network interface and source address.

Reply:
Got your point. That 's alternative way for static ip and default ip.